### PR TITLE
Some issues resolved

### PR DESCRIPTION
**Pull request by jJadsit**
Fixed issues #100 #111 and #103

### CODE REVIEW

**Review by Sean-Meyers — CHANGES_REQUESTED** [review]
2 Bugs found:
1) Opening the make some qrs activity seems to open the my codes activity instead, though, if you click back, it'll take you to the make some qrs activity.

2) After I scanned the first qr code and went to the my codes activity, everything displayed fine (nice job on the layout btw), but once I scanned 3 more codes and went to my codes activity, none of my qr codes showed up, although, on my profile and in the leaderboards it clearly shows that I have scanned 4 qr codes. Also, when I click on someone someone else's profile and view their qr codes, I can see all of their qr codes just fine, although the maximum number anyone else has right now is just 3, idk if that's a factor or not though.
Also, when I make a new account and try to view my qr codes or try to view my old account's qr codes from the leaderboard or whatever, I can't, for whatever reason, the view qr code activity broke after the first time I used it, and it remains broken for any account made while it was broken, yet is not broken for any account made when it was not broken.
From logcat, the only thing different between when manage codes seem to not work compared to when it works is that, when it doesn't work, database controller says it couldn't find the qr code id.
Idk how helpful my explanations are. I can show you in person if needed, assuming the bug will happen again, that is.

Otherwise, sorting in the leaderboards and stuff seems to work fine. sorting other people's qr codes worked fine too. Navigating to the make some qrs activity from the logout dialogue works fine, logout stuff otherwise works fine.

**Comment by jJadsit** [comment]
Fixed bug 1.
For the purpose of logging & actually using GitHub, Imma re-paste what I dm'd you here, in regards to bug 2

-   The second bug is hard to follow & understand, but you mention that DatabaseController says that it couldn't find ID's of some codes in firebase. The only time I've ever encountered this error message is from my local storage and firebase storage becoming desynched with each other, as a result of editing firebase data without resetting local data. This scenario leads to all sorts of unpredictable and numerous bugs, which sound like what you were reporting.
-   Do you think you could reset your local storage and try to replicate the bugs from a fresh state?

**Comment by jJadsit** [comment]
Just added more content to the PR that needs review. Sorry!
New stuff:

- #104 CodeScannerActivity split into CameraActivity and CodeSaveActivity. Should be easy to update image taking from here so that user has a choice to take a new image of the object they just scanned, and overall, should be easier to manage CameraActivity now that a lot of event based programming is gone
- Made a small change to ManageCodesActivity that may or may not fix some or all of the bugs you encountered. What was fixed for certain: If you have a QRCode on your local memory that is not in Firebase, all your scanned codes will load in the activity with the exception of the codes that are not in the remote.
- Put a fade out on the buttons in HubActivity. It doesn't look great but improving it will be as simple as editing the png files in the resource folder, no changes to the logic are required. This is a card in the roadmap, but not a full issue, so can't reference.

**Comment by jJadsit** [comment]
New stuff to enter Staging

- Updated some ugly/overlapping ui in some places (hopefully all, not sure)
- Fixed bug #120 (probably)
- Implemented feature #102, works now (i think)

**Review by Sean-Meyers — APPROVED** [review]
All the previous bugs that I found are fixed. Good job!

I didn't find any bugs from your work with splitting the code scanner activity into a code saving activity, but I think that we should put the button for saving a photo into the saving screen later, and make the button direct to a picture taking activity. It should be fine to repurpose the scanning activity into a picture taking activity with minimal changes for that. I didn't take too much of a look at the actually code in this pull request, so I'm not sure if the javadocs in the code scanner activity still make sense. I'll take a look at that later.

The fade buttons should probably be replaced with little arrows, like "<", and ">", This also avoids the need to think about dark mode probably.

Good job making the text in the profile viewer render better, long names no longer break the ui there.

Profile QR codes also work great, I think we should have a button that directs to the code scanner activity from the search players activity as well tho, so that it's intuitive that you can search for players by scanning a code.
